### ISSUE TITLE
[REF-1943] Backend variables should not be accessible to render

### ISSUE THREAD

**Issue by wassafshahzad** [reflex-dev/reflex#2559]
### Description
Working on the issue  #1154, we discovered that the backend variables are accessible to the frontend which they should not be.

###Steps to reproduce the behavior:
Run the following code
```
import random
import reflex as rx

class State(rx.State):

    _private: str = "Should work ?"


def index() -> rx.Component:
    return rx.vstack(
        rx.heading(f"Computed BE {State._private}"),
        spacing="1.5em",
        font_size="2em",
        padding_top="10%",
    )


app = rx.App()
app.add_page(index)
```

###Expected behavior
Backend vars should not be accessible to frontend

**Specifics (please complete the following information):**
 - Python Version: 3.12.1
 - Reflex Version: development version
 - OS: Windows
 - Browser (Optional):
 
**Additional context**
Linked PR #2540 

<sub>[REF-1943](https://linear.app/reflex-dev/issue/REF-1943/backend-variables-should-not-be-accessible-to-render)</sub>

**Comment by benedikt-bartscher** [comment]
@wassafshahzad your example code throws an exception: `TypeError: cannot pickle 'ComputedVar' object`. One could argue that this isn't a nice exception, but it does not seem like a bug.

**Comment by wassafshahzad** [comment]
> @wassafshahzad your example code throws an exception: `TypeError: cannot pickle 'ComputedVar' object`. One could argue that this isn't a nice exception, but it does not seem like a bug.

On the dev version ?

**Comment by wassafshahzad** [comment]
> @wassafshahzad your example code throws an exception: `TypeError: cannot pickle 'ComputedVar' object`. One could argue that this isn't a nice exception, but it does not seem like a bug.

I updated the example, I was working with his example on my local branch and it had some changes which prevented the above exception. Know the example should run

**Comment by benedikt-bartscher** [comment]
You are right, your example renders a backend-var to the frontend, which should not be possible. Tested with `main` branch and reflex v0.3.10

**Comment by wassafshahzad** [comment]
PR is up and awaiting review

**Comment by masenf** [comment]
PR seemed to have an issue #2681

**Comment by benedikt-bartscher** [comment]
@wassafshahzad I added some basic tests which fail against the problematic changes here: https://github.com/reflex-dev/reflex/pull/2685  
you can run them using `pytest integration -k test_state_in`

**Comment by wassafshahzad** [comment]
> @wassafshahzad I added some basic tests which fail against the problematic changes here: https://github.com/reflex-dev/reflex/pull/2685  
> you can run them using `pytest integration -k test_state_in`

Thank you, I will fiz it tomorrow or later today

**Comment by wassafshahzad** [comment]
> PR seemed to have an issue #2681 

Sorry, I will fix it later today

**Comment by masenf** [comment]
Tabling this one for now.

**Comment by adhami3310** [comment]
Since this only happens at compile time, and requires the developer to manually use a backend variable, this seems to be the thing of "why stop you if you're this dedicated"
